pkg/envinfo/envinfo.go

```diff
@@ -350,6 +353,17 @@ func (ei *EnvInfo) GetDebugPort() int {
 	return *ei.componentSettings.DebugPort
 }
 
+// GetUserCreatedDevfile returns the UserCreatedDevfile
+func (ei *EnvInfo) GetUserCreatedDevfile() bool {
```

Comment:
either way it doesn't make a difference? why do you feel a function makes more sense?
In reality 
```
 func (ei *EnvInfo) GetUserCreatedDevfile() bool
```
is more easy to use as one would just assume getting the value of `UserCreatedDevfile` but with function they would have to look for it and might just do `ei.UserCreateDevfile`

Reply:
> either way it doesn't make a difference? why do you feel a function makes more sense?

I feel this way because recently I have been reading about "methods vs functions in Go" and something I am reading often is that we should use a function if the state is not getting changed (which is the case here.) Using methods, for cases like this, is more OOP approach.

> is more easy to use as one would just assume getting the value of `UserCreatedDevfile` but with function they would have to look for it and might just do `ei.UserCreateDevfile`

I think that in either case, function body is going to stay the same, that is:
```go
return ei.componentSettings.UserCreatedDevfile
```
So I'm not sure how method would be easier than function.

Reply:
from an external user in method approach you just `. GetUserCreatedDevfile` for Function you need to go the package and see if there is a getter

Reply:
>  Function you need to go the package and see if there is a getter

It's not an inconvenience, per se. I think it's convenient OTOH. For method, let's say you have a variable named `ei` (taking it from function signature):
```go
ei.GetUserCreatedDevfile()
```
while if you use package name, it's more clear as to where this info is coming from:
```go
envinfo.GetUserCreatedDevfile()
```

Tangential comment - since it's a function returning boolean response, is `GetUserCreatedDevfile` the right name? That would indicate a function returning a user created devfile. IMO, `IsUserCreatedDevfile` is a better name.

Reply:
Yeah on variable name front agreed,
But I would ask - think from a new developer’s perspective,
How many things he needs to get UserCreateDevfile when using the method approach? How many in function approach?

in method approach he/she just needs to let the autocomplete help them and also it comes naturally that if the variable is in the struct you could be able to get it directly.

in function approach they would need to know the package name, function name, the argument that it takes? Doesn’t that feel like more work?

Reply:
Well, the IDEs would make this quite simple as far as knowing what the function needs. I can say that from whatever experience I have using VS Code & GoLand. But I agree with the comment you made on our Slack discussion about `UserCreatedDevfile` indicating the state. I should have realized that myself but it made more sense after you said it.

However, I still feel that we should name this "method" as `IsUserCreatedDevfile`. WDYT?

Reply:
done